runtime/alloc/alloc.c

```diff
@@ -3,177 +3,72 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <stdbool.h>
-#include "jemalloc/jemalloc.h"
 
 #include "runtime/alloc.h"
 #include "runtime/header.h"
+#include "runtime/arena.h"
 
-const size_t BLOCK_SIZE = 1024 * 1024;
 
-struct semispace_info {
-  char* first_block;
-  char* block;
-  char* block_start;
-  char* block_end;
-  char semispace_id;
-};
+REGISTER_ARENA(youngspace, 0);
+REGISTER_ARENA(oldspace, 1);
 
-static struct semispace_info fromspace;
-static struct semispace_info oldspace = {0, 0, 0, 0, 2};
-
-static char* first_tospace_block;
-
-char *fromspace_ptr() {
-  return fromspace.first_block;
+char *youngspace_ptr() {
+  return arenaStartPtr(&youngspace);
```

Comment:
have you checked that the assembly for these four functions do not contain a call instruction and that it gets successfully inlined?

Reply:
Yes I can verify that these are getting inlined properly. I checked the assembly code of the evm interpreter to make 100% sure.